docs/getting-started/walkthroughs/datastream_api.md

```diff
@@ -0,0 +1,897 @@
+
+<div class="codetabs" markdown="1">
+<div data-lang="java" markdown="1">
+{% highlight java %}
+alerts.addSink(new AlertSink());
+{% endhighlight %}
+</div>
+
+<div data-lang="scala" markdown="1">
+{% highlight scala %}
+alerts.addSink(new AlertSink)
+{% endhighlight %}
+</div>
+</div>
+
+#### Executing The Job
+
+Flink applications are built lazily and shipped to the cluster for execution only once fully formed.
+You can call `StreamExecutionEnvironment#execute` to begin the execution of our Job by giving it a name.
+
+<div class="codetabs" markdown="1">
+<div data-lang="java" markdown="1">
+{% highlight java %}
+env.execute("Fraud Detection");
+{% endhighlight %}
+</div>
+
+<div data-lang="scala" markdown="1">
+{% highlight scala %}
+env.execute("Fraud Detection")
+{% endhighlight %}
+</div>
+</div>
+
+#### The Fraud Detector
+
+The logic for the fraud detector is encapsulated within a `KeyedProcessFunction`.
+This first version outputs an alert on every transaction, which some may say is overly conservative.
+It also includes several constants that you may find helpful as you work through your implementation.
+
+<div class="codetabs" markdown="1">
+<div data-lang="java" markdown="1">
+{% highlight java %}
+public class FraudDetector extends KeyedProcessFunction<Long, Transaction, Alert> {
+
+    public static final double SMALL_AMOUNT = 0.01;
+
+    public static final double LARGE_AMOUNT = 500.00;
+
+    public static final long ONE_DAY = 24 * 60 * 60 * 1000;
+
+    @Override
+    public void processElement(
+        Transaction transaction,
+        Context context,
+        Collector<Alert> collector) throws Exception {
+  
+        Alert alert = new Alert();
+        alert.setId(transaction.getAccountId());
+
+        collector.collect(alert);
+    }
+}
+{% endhighlight %}
+</div>
+
+<div data-lang="scala" markdown="1">
+{% highlight scala %}
+object FraudDetector {
+
+    val SMALL_AMOUNT = 0.01
+
+    val LARGE_AMOUNT = 500.00
+
+    val ONE_DAY = 24 * 60 * 60 * 1000L
+}
+
+class FraudDetector extends KeyedProcessFunction[Long, Transaction, Alert] {
+
+    override def processElement(
+        transaction: Transaction,
+        context: Context,
+        collector: Collector[Alert]): Unit = {
+
+        Alert alert = new Alert
+        alert.setId(transaction.getAccountId)
+
+        collector.collect(alert)
+    }
+}
+{% endhighlight %}
+</div>
+</div>
+
+## Writing An Initial Application 
+
+For the initial implementation, the fraud detector should output an alert for any account that makes a small transaction immediately followed by a large one. Where small is anything less than $0.10 and large is more than $500.
+Imagine your fraud detector processes the following stream of transactions for a particular account.
+
+<p class="text-center">
+    <img alt="Transactions" width="80%" src="{{ site.baseurl }}/fig/fraud-transactions.svg"/>
```

Comment:
Can you enlarge the non-fraud arrow so that it actually covers the three tx you are talking about?